src/translate.service.ts

```diff
@@ -230,7 +231,8 @@ export class TranslateService {
 
         // if this language is unavailable, ask for it
         if(typeof this.translations[lang] === "undefined") {
-            pending = this.getTranslation(lang);
+            this._translationRequests[lang] = this._translationRequests[lang] || this.getTranslation(lang);
```

Comment:
I can use a nested if here to only set it to this.getTranslation() if it is undefined. Let me know if you prefer that.